conductor-client/src/main/kotlin/com/openlattice/postgres/external/ExternalDatabasePermissioner.kt

```diff
@@ -0,0 +1,515 @@
+                    stmt.execute(createRoleSql)
+                    stmt.execute(grantSchemaSql)
+                    stmt.execute(grantViewSql)
+                }
+            }
+        }
+    }
+
+    /**
+     * Updates permissions on [propertyTypes] for [entitySet] in org database for [organizationId]
+     */
+    override fun updateAssemblyPermissions(
+            action: Action,
+            columnAcls: List<Acl>,
+            columnsById: Map<AclKey, TableColumn>
+    ) {
+        updateTablePermissions(action, columnAcls, columnsById, TableType.VIEW)
+    }
+
+    /**
+     * Create all postgres roles to apply to [table] and [columns] in [organizationId] database
+     */
+    override fun initializeExternalTablePermissions(
+            organizationId: UUID,
+            table: OrganizationExternalDatabaseTable,
+            columns: Set<OrganizationExternalDatabaseColumn>
+    ) {
+        val sqlCommandsPerPropertyType = columns.map { col ->
+            val quotedColumn = ApiHelpers.dbQuote(col.name)
+            olToPostgres.map { (permission, pgPermission) ->
+                val roleName = PostgresRoles.getOrCreatePermissionRole( externalRoleNames, permission, table.id, col.id)
+                val pgPermissionString = pgPermission.joinToString()
+                createRoleIfNotExistsSql(roleName) to grantPermissionsOnColumnsOnTableToRoleSql(
+                        pgPermissionString,
+                        quotedColumn,
+                        null,
+                        table.name,
+                        roleName
+                )
+            }
+        }
+
+        extDbManager.connectToOrg(organizationId).connection.use { conn ->
+            conn.createStatement().use { stmt ->
+                sqlCommandsPerPropertyType.forEach { createSqlToGrantSql ->
+                    createSqlToGrantSql.map { (createRoleSql, grantSql) ->
+                        stmt.addBatch(createRoleSql)
+                        stmt.addBatch(grantSql)
+                    }
+                    stmt.executeBatch()
+                }
+            }
+        }
+    }
+
+    /**
+     * Updates permissions on [columns] for [table] in org database for [organizationId]
+     */
+    override fun updateExternalTablePermissions(
+            action: Action,
+            columnAcls: List<Acl>,
+            columnsById: Map<AclKey, TableColumn>
+    ) {
+        updateTablePermissions(action, columnAcls, columnsById, TableType.TABLE)
+    }
+
+    private fun updateTablePermissions(
+            action: Action,
+            columnAcls: List<Acl>,
+            columnsById: Map<AclKey, TableColumn>,
+            tableType: TableType
+    ) {
+        if (action != Action.ADD && action != Action.REMOVE && action != Action.SET) {
+            logger.error("Action $action passed through to updateTablePermissions is unhandled. Doing no operations")
+            return
+        }
+
+        val allOrgs = columnsById.values.map { it.organizationId }
```

Comment:
probably want a `.toSet()` since you're iterating over this and doing work for each one